jena-arq/src/main/java/org/apache/jena/sparql/engine/iterator/QueryIterLateral.java

```diff
@@ -114,7 +124,7 @@ public Op transform(OpBGP opBGP) {
             Op opExec = substitute
                     ? Substitute.substitute(opBGP, substitutions)
                     : opBGP;
-            opExec = OpAssign.create(opExec, assigns);
+            opExec = OpExtend.create(opExec, assigns);
```

Comment:
For consistency also call `assignments(opExec, assigns)` instead?

Reply:
Thanks for catching that. It was change from "assign" to "extend" in the investigation phase of the report!